modin/numpy/arr.py

```diff
@@ -704,50 +741,99 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=None, *, where=True):
             if dtype is not None
             else (out.dtype if out is not None else self.dtype)
         )
+        out_type = out_dtype if not hasattr(out_dtype, "type") else out_dtype.type
+        if isinstance(where, array) and issubclass(out_type, numpy.integer):
+            out_dtype = numpy.float64
         check_kwargs(keepdims=keepdims, where=where)
         if self._ndim == 1:
             if axis == 1:
                 raise numpy.AxisError(1, 1)
-            result = self._query_compiler.astype(
-                {col_name: out_dtype for col_name in self._query_compiler.columns}
+            result = self._query_compiler
+            if isinstance(where, array):
+                result = where.where(self, numpy.nan)._query_compiler
+            result = result.astype(
+                {col_name: out_dtype for col_name in result.columns}
             ).mean(axis=0)
             if keepdims:
                 if out is not None and out.shape != (1,):
                     raise ValueError(
                         f"operand was set up as a reduction along axis 0, but the length of the axis is {out.shape[0]} (it has to be 1)"
                     )
-                return fix_dtypes_and_determine_return(result, 1, dtype, out, where)
+                if out is not None:
+                    out._query_compiler = (
+                        numpy.ones_like(out) * numpy.nan
+                    )._query_compiler
+                return (
+                    fix_dtypes_and_determine_return(
+                        result, 1, dtype, out, where is not False
+                    )
+                    if where is not False or out is not None
+                    else array([numpy.nan], dtype=out_dtype)
+                )
             return result.to_numpy()[0, 0] if where else numpy.nan
```

Comment:
shouldn't we change this to `if where is not False` as well?

Reply:
No - in this case, where should just be evaluated for truthiness, since we're returning a single value.

Reply:
I wonder if this should be a comment in the code, as this is the only place where the pattern is different...

Reply:
Sure - I can add a comment!